packages/hardhat-core/src/internal/util/bigint.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  BigIntLike,
+  bufferToBigInt,
+  bufferToHex,
+  toBuffer,
+} from "@ethereumjs/util";
+
+function min(x: bigint, y: bigint): bigint {
+  return x < y ? x : y;
+}
+
+function max(x: bigint, y: bigint): bigint {
+  return x > y ? x : y;
+}
+
+function isBigInt(x: unknown): x is bigint {
+  return typeof x === "bigint";
+}
+
+function divUp(x: bigint, y: bigint): bigint {
+  let result = x / y;
+
+  if (x % y !== 0n) {
+    result = result + 1n;
+  }
+
+  return result;
+}
+
+function cmp(a: bigint, b: bigint): number {
+  return a < b ? -1 : a > b ? 1 : 0;
+}
+
+function mapNumberToBigInt<T>(x: number | T): bigint | T {
+  if (typeof x === "number") {
+    return BigInt(x);
```

Comment:
Shouldn't we validate this?

Reply:
I can't find tests for this module. That makes me think that it's even more important to validate this, as it's unclear what happens if there's an overflow.

Reply:
PS: These comments were for the other function, the one turning bigints into numbers 😂

Reply:
Oh yes, we need tests for this module, my bad.

Reply:
Added tests for this module in https://github.com/NomicFoundation/hardhat/pull/3082/commits/2ee5e5e5741ba0731d49c5a5859a03962a40fa2f